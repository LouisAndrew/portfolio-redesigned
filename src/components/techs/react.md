---
templateKey: techs
title: React
name: ReactJS
icon: assets/react-1-logo-png-transparent.png
---
