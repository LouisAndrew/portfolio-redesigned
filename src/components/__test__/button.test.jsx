import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'

import Button from '../button'

afterEach( cleanup )

it('renders without crashing', () => {

        const div = document.createElement('div')
        ReactDOM.render(<Button>Test Button</Button>, div )
})

it('renders button correctly', () => {

        const text = 'Test'
        const { getByTestId } = render( <Button>{text}</Button> )
        expect( getByTestId('button') ).toHaveTextContent(text)
})

it('matches snapshot', () => {

        const ref = React.createRef()
        const text = 'Test'
        const tree = renderer.create( <Button ref={ref}>{text}</Button> ).toJSON()
        expect(tree).toMatchSnapshot()
})