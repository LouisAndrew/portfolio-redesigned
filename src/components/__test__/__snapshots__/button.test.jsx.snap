// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="sc-AxjAm hNoFXg"
  color="def"
  data-testid="button"
  style={Object {}}
>
  Test
</button>
`;
